### PR TITLE
added HIP backend for activation layer

### PR DESCRIPTION
**Pull request by rrawther**
Added HIP backend option for activation layer which uses MIOpen

### CODE REVIEW

**Comment by rrawther** [comment]
@kiritigowda : merge this after merging PR#575

**Review by asalmanp — APPROVED** [review]

**Comment by kiritigowda** [comment]
CI set - http://math-ci.rocm.amd.com/blue/organizations/jenkins/compute-rocm-dkms-no-npi-hipclang%2FShort-GPUOpen%2FMIVisionX/detail/PR-576/1/pipeline

**Pull request merged**
